internal/receiver/simpleprometheusremotewritereceiver/reporter.go

```diff
@@ -0,0 +1,107 @@
+// Copyright Splunk, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package simpleprometheusremotewritereceiver
+
+import (
+	"context"
+
+	"go.opencensus.io/trace"
+	"go.opentelemetry.io/collector/obsreport"
+	"go.opentelemetry.io/collector/receiver"
+	"go.uber.org/zap"
+
+	"github.com/signalfx/splunk-otel-collector/internal/receiver/simpleprometheusremotewritereceiver/internal"
+)
+
+var _ internal.Reporter = (*reporter)(nil)
+
+// reporter struct implements the transport.Reporter interface to give consistent
+// observability per Collector metric observability package.
+type reporter struct {
+	logger        *zap.Logger
+	sugaredLogger *zap.SugaredLogger // Used for generic debug logging
+	obsrecv       *obsreport.Receiver
+}
+
+func newReporter(settings receiver.CreateSettings) (internal.Reporter, error) {
+	obsrecv, err := obsreport.NewReceiver(obsreport.ReceiverSettings{
+		ReceiverID:             settings.ID,
+		Transport:              "tcp",
+		ReceiverCreateSettings: settings,
+	})
+	if err != nil {
+		return nil, err
+	}
+	return &reporter{
+		logger:        settings.Logger,
+		sugaredLogger: settings.Logger.Sugar(),
+		obsrecv:       obsrecv,
+	}, nil
+}
+
+// OnDataReceived is called when a message or request is received from
+// a client. The returned context should be used in other calls to the same
+// reporter instance. The caller code should include a call to end the
+// returned span.
+func (r *reporter) OnDataReceived(ctx context.Context) context.Context {
```

Comment:
Looks like we're not using this method, but the comments state that the returned context of this method should be passed into the other methods.

Reply:
Personally, I would leave out the obsreport stuff altogether and save it for its own PR once the receiver's basic functionality is merged.